src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseGradingService.java

```diff
@@ -167,6 +163,31 @@ public Optional<Result> processNewProgrammingExerciseResult(@NotNull Programming
         return Optional.ofNullable(newResult).map(result -> processNewProgrammingExerciseResult(participation, result));
     }
 
+    /**
+     * Checks that the build result belongs to the default branch of the exercise.
+     *
+     * @param exercise The exercise in which the submission was made.
+     * @param buildResult The build result received from the CI system.
+     * @throws IllegalArgumentException Thrown if the result does not belong to the default branch of the exercise.
+     */
+    private void checkCorrectBranchElseThrow(final ProgrammingExercise exercise, final AbstractBuildResultNotificationDTO buildResult) throws IllegalArgumentException {
+        final String exerciseDefaultBranch = exercise.getBranch();
```

Comment:
Are you sure that the branch is correctly stored in the exercise? I'd suggest using getOrRetrieveBranchOfExercise() just to make sure

Reply:
Good idea, I wasn’t aware of that method. :+1: I updated the check.